Yosemite/Yosemite/Stores/ProductCategoryStore.swift

```diff
@@ -52,7 +52,7 @@ private extension ProductCategoryStore {
             }
 
             self?.upsertStoredProductCategoriesInBackground(productCategories, siteID: siteID) {
-                onCompletion(nil)
+                onCompletion(productCategories, nil)
```

Comment:
I was having a play with this and found little bit complicated to return the actual stored objects without having to make an extra core-data call, since the `upsert` and `save` happen in different closures.

```swift
derivedStorage.perform { [weak self] in
    self?.upsertStoredProductCategories(readOnlyProductCategories, in: derivedStorage, siteID: siteID)
}

storageManager.saveDerivedType(derivedStorage: derivedStorage) {
    DispatchQueue.main.async(execute: onCompletion)
}
```

And at that point it seems like a little too-much effort 😅 
Would you mind if I add an item to tackle this later or do you feel strongly about it?

Reply:
Thanks for taking a look at it. Let's leave it as is since it's too much effort. I didn't realize that it was. My bad. 😅  

It's at least good to know that this could cause some mysterious bugs in the future if the `Mapper` implementations do not match the `toReadOnly()` (reverse) implementations.